crd-generator/src/main/java/io/fabric8/crd/generator/apt/CustomResourceAnnotationProcessor.java

```diff
@@ -0,0 +1,145 @@
+import com.fasterxml.jackson.dataformat.yaml.YAMLGenerator;
+import com.fasterxml.jackson.dataformat.yaml.YAMLGenerator.Feature;
+import io.fabric8.crd.generator.v1.CustomResourceHandler;
+import io.fabric8.kubernetes.api.model.KubernetesList;
+import io.fabric8.kubernetes.client.CustomResource;
+import io.fabric8.kubernetes.client.utils.ApiVersionUtil;
+import io.sundr.codegen.CodegenContext;
+import io.sundr.codegen.functions.ElementTo;
+import io.sundr.codegen.model.TypeDef;
+import java.io.IOException;
+import java.io.Writer;
+import java.util.LinkedHashSet;
+import java.util.List;
+import java.util.Set;
+import javax.annotation.processing.AbstractProcessor;
+import javax.annotation.processing.RoundEnvironment;
+import javax.annotation.processing.SupportedAnnotationTypes;
+import javax.lang.model.element.Element;
+import javax.lang.model.element.Name;
+import javax.lang.model.element.TypeElement;
+import javax.lang.model.type.DeclaredType;
+import javax.lang.model.type.TypeMirror;
+import javax.tools.FileObject;
+import javax.tools.StandardLocation;
+
+@SupportedAnnotationTypes({"io.fabric8.kubernetes.model.annotation.Version"})
+public class CustomResourceAnnotationProcessor extends AbstractProcessor {
+
+  private static final ObjectMapper YAML_MAPPER = new ObjectMapper(
+    new YAMLFactory()
+      .enable(Feature.MINIMIZE_QUOTES)
+      .enable(Feature.ALWAYS_QUOTE_NUMBERS_AS_STRINGS)
+      .disable(YAMLGenerator.Feature.WRITE_DOC_START_MARKER)
+  );
+  static {
+    YAML_MAPPER.configure(SerializationFeature.INDENT_OUTPUT, true);
+    YAML_MAPPER.configure(SerializationFeature.WRITE_NULL_MAP_VALUES, false);
+    YAML_MAPPER.configure(SerializationFeature.WRITE_EMPTY_JSON_ARRAYS, false);
+  }
+
+  private static final String CUSTOM_RESOURCE_NAME = CustomResource.class.getCanonicalName();
+  private final Resources resources = new Resources();
+  private final CustomResourceHandler v1Handler = new CustomResourceHandler(resources);
+  private final io.fabric8.crd.generator.v1beta1.CustomResourceHandler v1beta1Handler = new io.fabric8.crd.generator.v1beta1.CustomResourceHandler(
+    resources);
+
+  public boolean process(Set<? extends TypeElement> annotations, RoundEnvironment roundEnv) {
+    if (roundEnv.processingOver()) {
+      // write files
+      final KubernetesList list = resources.generate();
+      list.getItems().forEach(crd -> {
+        final String version = ApiVersionUtil.trimVersion(crd.getApiVersion());
+        try {
+          FileObject yml = processingEnv.getFiler()
+            .createResource(StandardLocation.CLASS_OUTPUT, "",
+              "META-INF/fabric8/" + crd.getMetadata().getName() + "-" + version + ".yml");
+          try (Writer writer = yml.openWriter()) {
+            final String yamlValue = YAML_MAPPER.writeValueAsString(crd);
+            writer.write(yamlValue);
+            writer.flush();
+            System.out.println("Created " + yml.toUri());
```

Comment:
Can we avoid using `System.out.println` in this class? Is it intentional?

Reply:
It's intentional so that the user can know which files are generated and where. What would you use instead of `System.out` since I'm not sure that loggers will actually work during the annotation processing phase?

Reply:
This is indeed intentional to be able to show which files were generated to the user. What alternative would you use for this purpose since I don't think that loggers work at compile time?